jwala-services/src/main/java/com/cerner/jwala/service/media/impl/MediaServiceImpl.java

```diff
@@ -97,12 +105,13 @@ public void remove(final String name) {
 
     /**
      * This method will check for the existing jvm associations for the media
+     *
      * @param name media name
      */
     private void checkForJvmAssociation(String name) {
         List<Jvm> jvmList = jvmPersistenceService.getJvms();
         for (Jvm jvm : jvmList) {
-            if (name.equalsIgnoreCase(jvm.getJdkMedia().getName())) {
+            if (jvm.getJdkMedia() != null && name.equalsIgnoreCase(jvm.getJdkMedia().getName())) {
```

Comment:
Can you add the JVM name(s) in the message? Add the JVM Names in a List and throw an exception outside the loop. This will help the users identify the jvm associated with the media and delete it or change them if required.